jscomp/syntax/src/res_core.ml

```diff
@@ -2714,9 +2714,10 @@ and parseJsxFragment p =
   Parser.expect GreaterThan p;
   let _spread, children = parseJsxChildren p in
   let childrenEndPos = p.Parser.startPos in
+  if p.token = LessThan then p.token <- Scanner.reconsiderLessThan p.scanner;
   Parser.expect LessThanSlash p;
-  Parser.expect GreaterThan p;
   Scanner.popMode p.scanner Jsx;
+  Parser.expect GreaterThan p;
```

Comment:
we first check if the next token is not `</` instead of just `<` to test the end of the fragment with a following `>` (forming a `</>`), if it's the case then we can pop JSX mode.